facebook/react-native
# Implement Blob support for XMLHttpRequest

This PR is a followup to https://github.com/facebook/react-native/pull/11417 and should be merged after that one is merged.

### What does the PR do?
1. Add support for creating blobs from strings, not just other blobs
2. Add the `File` constructor which is a superset of `Blob`
3. Add the `FileReader` API which can be used to read blobs as strings or data url (base64)
4. Add support for uploading and downloading blobs via `XMLHttpRequest` and `fetch`
5. Add ability to download local files on Android so you can do `fetch(uri).then(res => res.blob())` to get a blob for a local file (iOS already supported this)

### Test plan
1. Clone the repo https://github.com/expo/react-native-blob-test
2. Change the `package.json` and update `react-native` dependency to point to this branch, then run `npm install`
3. Run the `server.js` file with `node server.js`
4. Open the `index.common.js` file and replace `localhost` with your computer's IP address
5. Start the packager with `yarn start` and run the app on your device

If everything went well, all tests should pass, and you should see a screen like this:

![screen shot 2017-06-08 at 7 53 08 pm](https://user-images.githubusercontent.com/1174278/26936407-435bbce2-4c8c-11e7-9ae3-eb104e46961e.png)!

Pull to rerun all tests or tap on specific test to re-run it

## Release Notes
 [GENERAL] [FEATURE] [Blob] - Implement blob support for XMLHttpRequest
